### PR TITLE
Removed beginning underscore from some setters/getters

### PR DESCRIPTION
**Pull request by YeldhamDev**
Just for consistency sake, as they seemed like the only ones that had them.

Also renamed OptionButton's "_select_int" to "set_selected", to match the getter.

### CODE REVIEW

**Comment by groud** [comment]
The underscore before the function name makes it unavailable from GDscript (thus it make the function  editor-only). Thus it may have been intended.

**Comment by vnen** [comment]
As I commented on IRC, I'm not sure if the setgets should be made visible or if it's the properties that should be hidden. But I don't see the point of making a visible property with a hidden setter/getter pair.

**Comment by NathanWarden** [comment]
Unless I'm missing something, I believe it was done that way because that's how properties typically work in other languages. It's essentially two methods (get and set) that are hidden by something that looks like a field, but really isn't.

IE: node.visible is actually two methods that are something like: `node._set_visible(bool visible)` and `bool node._get_visible()`.

In C# you can make a property:

``` C#
public bool visible { get; set; }
```

Which internally looks something like: `get_visible()` and `set_visible()` but is simply called by `whatever.visible`.

In other words, I'm 99% sure the methods are intentionally hidden so that the person doing the scripting only has to worry about a single property as opposed to two methods. Making these visible now makes it a property and two methods instead of single property which does the same thing as the two methods :)

**Comment by vnen** [comment]
@NathanWarden not really, because setters and getters are shown with the property in editor help:

godot windows opt tools 64_2017-11-25_14-58-26

They aren't shown in the regular list of methods.

OTOH, for those underscore setters and getters, they are still shown along the properties, even though non-setget methods are hidden if they start with an underscore. For instance, the MenuButton `items` property:

godot windows opt tools 64_2017-11-25_14-59-53

----

My guess is that those setgets were supposed to be hidden but ended up in the surface when properties were added. That's why I'm wondering if those properties were meant to be visible or even if the should be bound at all. E.g. instead of using `MenuButton.items`, you should use `MenuButton.get_popup().add_item()`, so the proper structure is generated.

**Comment by NathanWarden** [comment]
From the documentation instructions at: https://hackmd.io/s/H1es51Oeqb#

DO NOT EDIT SET/GET methods! They’ll be removed from the docs in Godot 3.0. Instead, document the member variables (anything in the <members></members> XML nodes)

So, they're definitely supposed to be hidden.

**Comment by reduz** [comment]
Those should not be visible in the doc, it's more of a doc bug

**Comment by akien-mga** [comment]
@NathanWarden No, those doc instructions are outdated.

**Comment by vnen** [comment]
@reduz should those specific properties in this PR be available to GDScript? Either the properties should be made hidden or the setters should be made visible.

**Comment by YeldhamDev** [comment]
@groud They can be used in GDScript.

**Comment by NathanWarden** [comment]
For the methods that make sense to have a property, they should either be all exposed properties or all exposed methods, but not both. Obviously if we're keeping methods we just simply shouldn't use properties. If we're using properties, then the related methods should be hidden.

@akien-mga @NathanLovato Are you sure it's outdated? If that part of it is outdated then it should probably be removed, right?

**Comment by NathanLovato** [comment]
@NathanWarden Yes it was changed... yesterday? The warning was because you would have a lot of setters and getters that were bound to a member variable and ignored while building the docs - but docs writers still filled them. The thing is that was not *all* of the methods starting with set/get, just the majority.
All setters and getters are gone since... yesterday? Now the remaining methods that start with set/get are different, they're not bound to variables.

**Comment by NathanWarden** [comment]
@NathanLovato Ah, okay, I think that makes sense if I'm understanding correctly... All of the setters/getters for properties are gone, but the rest are still there. Am I understanding that correctly? Thanks :)

**Comment by reduz** [comment]
I think the proper way to solve this problem would be to:
1. add a PROPERTY_USAGE_INTERNAL mask in object.h
2. add it to the usage of the properties that we dont want visible
3. make sure they are not used in doc_data.cpp and not used on bindings.

**Comment by akien-mga** [comment]
I've opened #13334 to keep track of the issue, closing this PR as it's not the right fix.
